app/controllers/api/v2/insights_advisor/insights_advisor_controller.rb

```diff
@@ -0,0 +1,78 @@
+module Api
+  module V2
+    module InsightsAdvisor
+      class InsightsAdvisorController < ::Api::V2::BaseController
+        include ::Api::Version2
+        include Foreman::Controller::SmartProxyAuth
+        add_smart_proxy_filters [:upload_hits]
+
+        api :PATCH, "insights_advisor/upload_hits", N_("Upload from insights advisor")
+        param :host_name, String, required: true
+        param :host_uuid, String, required: true
+
+        param :payload, Hash, :desc => N_("On prem payload including resolutions, rules, hits") do
```

Comment:
can we configure that one not to be logged? it's *massive*

Reply:
Who's the Rails Black Belt here?! ;-)

My `git grep` suggests it should look like this: https://github.com/Katello/katello/blob/master/app/controllers/katello/api/v2/content_uploads_controller.rb#L9
```
filter_parameters :payload
```

Reply:
I added
```ruby
        include ::Foreman::Controller::FilterParameters
        filter_parameters :payload
```
and got
```
2025-01-14T02:27:19 [I|app|c9715072] Started PATCH "/api/insights_advisor/upload_hits" for 10.0.168.162 at 2025-01-14 02:27:19 -0500
2025-01-14T02:27:19 [I|app|c9715072] Processing by Api::V2::InsightsAdvisor::InsightsAdvisorController#upload_hits as JSON
2025-01-14T02:27:19 [I|app|c9715072]   Parameters: {"host_name"=>"ip-10-0-167-58.example.com", "host_uuid"=>"961989e3-58af-4424-8bc2-981c2ca3486c", "payload"=>"[FILTERED]", "apiv"=>"v2", "insights_advisor"=>{"host_name"=>"ip-10-0-167-58.example.com", "host_uuid"=>"961989e3-58af-4424-8bc2-981c2ca3486c", "payload"=>"[FILTERED]"}}
```

Reply:
updated